geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionImpl.java

```diff
@@ -72,8 +79,11 @@
 import org.apache.geode.internal.serialization.SerializationContext;
 import org.apache.geode.internal.serialization.Version;
 import org.apache.geode.internal.tcp.ConnectExceptions;
+import org.apache.geode.internal.tcp.ConnectionException;
 import org.apache.geode.internal.util.Breadcrumbs;
 import org.apache.geode.logging.internal.executors.LoggingThread;
+import org.apache.geode.security.AuthenticationRequiredException;
+import org.apache.geode.security.GemFireSecurityException;
 
 public class DistributionImpl implements Distribution {
```

Comment:
It'd be valuable to have a class-level comment block explaining the purpose of this class. One particular thing to note, is the fact that _all_ translation of membership exceptions to `geode-core` ones happens here.